hpctestlib/apps/amber/nve.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2016-2021 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+class Amber_NVE(rfm.RunOnlyRegressionTest, pin_prefix=True):
+    '''Base class for the Amber NVE Test.
+
+    Amber is a suite of biomolecular simulation programs. It
+    began in the late 1970's, and is maintained by an active
+    development community (see ambermd.org).
+
+    The presented abstract run-only class checks the amber perfomance.
+    To do this, it is necessary to define in tests the name
+    of the running script (input file), the output file,
+    as well as set the reference values of energy and possible
+    deviations from this value. This data is used to check if
+    the task is being executed correctly, that is, the final energy
+    is correct (approximately the reference). The default assumption
+    is that Amber is already installed on the device under test.
+    '''
+
+    #: Amber output file.
+    #:
+    #: :default: : 'amber.out'
+    output_file = variable(str, value='amber.out')
+
+    #: Amber input file. This file is set by the post-init hook
+    #: :func:`unpack_platform_parameter`.
+    #:
+    #: :default: :class:`required`
+    input_file = variable(str)
+
+    #: Reference value of energy, that is used for the comparison
+    #: with the execution ouput on the sanity step. The absolute
+    #: difference between final energy value and reference value
+    #: should be smaller than energy_tolerance
+    #:
+    #: :type: float
+    #: :default: :class:`required`
+    energy_value = variable(float)
+
+    #: Maximum deviation from the reference value of energy,
+    #: that is acceptable.
+    #:
+    #: :type: float
+    #: :default: :class:`required`
+    energy_tolerance = variable(float)
+
+    #: Parameter pack containing the platform ID, input file and
+    #: executable.
+    platform_info = parameter([
```

Comment:
These parameters pack lead to ugly names (at least for the moment). What's the purpose of these particular packs apart from grouping everything together? What if we had this as `platform` and set the executable and input file accordingly in the hook. Similarly with the `variant`.

Reply:
Well, this format makes it very easy to get ` input_file` and `executable` values in a hook, literally on 1 line. If we do without it, my understanding is that there is only an option with `if` loops, which is as ugly as possible.